File: 2350-find-closest-number-to-zero/find-closest-number-to-zero.cpp
class Solution {
public:
    int findClosestNumber(vector<int>& nums) {
        int minIndex = 0;
        for(int i = 0; i<nums.size(); i++){
            if (abs(nums.at(i)) < abs(nums.at(minIndex))){
                minIndex = i;
            }
            else if (abs(nums.at(i)) == abs(nums.at(minIndex)) && nums.at(i) > nums.at(minIndex)){
                minIndex = i;
            }
        }
        return nums.at(minIndex);
    }
};

File: 2350-find-closest-number-to-zero/find-closest-number-to-zero.py
import sys
class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return None
        
        min_index = 0
        for index, num in enumerate(nums):
            if abs(num) < abs(nums[min_index]):
                    min_index = index
            elif abs(num) == abs(nums[min_index]):
                if num > nums[min_index]:
                    min_index = index
        
        return nums[min_index]